packages/themes/build.js

```diff
@@ -0,0 +1,243 @@
+  Object.keys(obj).forEach((key) => {
+    obj[key] = replaceStringInMode(obj[key], mode, oldStr, newStr);
+  });
+
+  return obj;
+};
+
+// This function renames keys in the object based on the keyMap
+const renameKeysDeep = (obj, keyMap) => {
+  if (typeof obj !== 'object' || obj === null) {
+    return obj;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => renameKeysDeep(item, keyMap));
+  }
+
+  const renamedObj = {};
+  Object.keys(obj).forEach((key) => {
+    const newKey = keyMap[key] || key;
+    renamedObj[newKey] = renameKeysDeep(obj[key], keyMap);
+  });
+  return renamedObj;
+};
+
+StyleDictionary.registerParser({
+  pattern: /\.json$/,
+  parse: ({ contents }) => {
+    try {
+      const jsonObj = JSON.parse(contents);
+
+      // Replace the references to modes
+      const lightParsed = replaceStringInMode(
+        jsonObj['variables'],
+        'light-mode',
+        'semantic.dark-mode', // Reference to incorrect mode prior to replacement
+        'theme.light'
+      );
+
+      const darkParsed = replaceStringInMode(
+        lightParsed,
+        'dark-mode',
+        'semantic.dark-mode',
+        'theme.dark'
+      );
+
+      const parsedJsonObj = renameKeysDeep(darkParsed, {
+        semantic: 'theme',
+        'light-mode': 'light',
+        'dark-mode': 'dark'
+      });
+
+      return parsedJsonObj;
+    } catch (error) {
+      console.error(error);
+    }
+  }
+});
+
+console.log('Build started...');
+console.log('\n==============================================');
+
+/*
+ THESE ARE CONTITIONS TO FILTER ON AND A HELPER FUNCTION TO CREATE MATCHING FILTERS
+ These are a convenience to avoid long filter strings such as
+ token.attributes.category === 'global' || token.attributes.category === 'spacing' || token....
+ The objects defined here are allowed through - everything else is filtered out. Acts as a yes gate
+ The not: array allows for partial filtering of object keys. Acts as a no gate
+*/
+const conditionsCssLightTheme = [{ category: 'theme', type: 'light' }];
+
+const conditionsCssDarkTheme = [{ category: 'theme', type: 'dark' }];
+
+const conditionsCssGlobal = [
+  // token.attributes.category === 'global' || token.attributes.category === 'spacing'
+  { category: 'global', subitem: { not: ['seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsTw = [
+  { category: 'theme', type: 'light' },
+  { category: 'global', subitem: { not: ['light', 'dark', 'seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsJs = [
+  { category: 'theme' },
+  {
+    category: 'global',
+    type: { not: ['spacing'] },
+    item: { not: ['mode'] },
+    subitem: { not: ['seed'] }
+  }
+];
+
+const matchesCondition = (token, condition) => {
+  return Object.entries(condition).every(([key, value]) => {
+    if (typeof value === 'object' && value.not) {
+      return !value.not.includes(token.attributes[key]);
+    }
+    return token.attributes[key] === value;
+  });
+};
+
```

Comment:
Renaming `[key, value]` to `[field, filter]` and extracting a `val` constant is possibly slightly clearer:

```suggestion
const matchesCondition = (token, condition) => {
  return Object.entries(condition).every(([field, filter]) => {
    const val = token.attributes[field];
    
    if (typeof filter === 'object' && filter.not) {
      return !filter.not.includes(val);
    }
    return filter === val;
  });
};
```